pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -57,16 +53,11 @@ func (h *Helper) GetDockerAuth(imageName, authType string) (docker.AuthConfigura
 		return docker.AuthConfiguration{}, false
 	}
 
-	var cfg credentialprovider.DockerConfig
-	var err error
-	if strings.HasSuffix(dockercfgPath, kapi.DockerConfigJsonKey) || strings.HasSuffix(dockercfgPath, "config.json") {
-		cfg, err = readDockerConfigJson(dockercfgPath)
-	} else if strings.HasSuffix(dockercfgPath, kapi.DockerConfigKey) {
-		cfg, err = readDockercfg(dockercfgPath)
-	}
-
+	dockercfgDir, dockercfgFile := filepath.Split(dockercfgPath)
+	credentialprovider.SetPreferredDockercfgPath(dockercfgDir)
```

Comment:
Well the good news is we're doing all this in a container, so we're pretty
isolated. But I agree, maybe the best thing to do here is an upstream PR to
k8s to improve the usability of the config reader function before trying to
leverage it here.

Ben Parees | OpenShift

On Aug 29, 2016 1:02 AM, "Jordan Liggitt" <notifications@github.com> wrote:

> In pkg/build/builder/cmd/dockercfg/cfg.go
> <https://github.com/openshift/origin/pull/10608#discussion_r76553462>:
>
> > @@ -57,16 +53,11 @@ func (h *Helper) GetDockerAuth(imageName, authType string) (docker.AuthConfigura
> >  		return docker.AuthConfiguration{}, false
> >  	}
> >
> > -	var cfg credentialprovider.DockerConfig
> > -	var err error
> > -	if strings.HasSuffix(dockercfgPath, kapi.DockerConfigJsonKey) || strings.HasSuffix(dockercfgPath, "config.json") {
> > -		cfg, err = readDockerConfigJson(dockercfgPath)
> > -	} else if strings.HasSuffix(dockercfgPath, kapi.DockerConfigKey) {
> > -		cfg, err = readDockercfg(dockercfgPath)
> > -	}
> > -
> > +	dockercfgDir, dockercfgFile := filepath.Split(dockercfgPath)
> > +	credentialprovider.SetPreferredDockercfgPath(dockercfgDir)
>
> Really not a fan of mutating a global here, even temporarily. That means
> this function can't be used in a thread safe way
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10608/files/af73cdd0548ee2c42e05192b75e1e43f91a6de47#r76553462>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEvl3l8lsMkbhh2MoASAWVa-RrcRiravks5qkmfagaJpZM4JrsxA>
> .
>